lib/stdlib/test/peer_SUITE.erl

```diff
@@ -0,0 +1,463 @@
+    Conn = proplists:get_value(connection, Config),
+    {ok, Peer, _Node} = peer:start_link(#{connection => Conn}),
+    ct:capture_start(),
+    peer:call(Peer, io, format, [user, "test.", []]),
+    peer:call(Peer, erlang, apply, [io, format, ["second."]]),
+    %% ensure no dist connection is set up
+    ?assertNot(lists:member(Peer, nodes()), {dist_connected, Peer, nodes()}),
+    ct:capture_stop(),
+    Texts = ct:capture_get(),
+    peer:stop(Peer),
+    ?assertEqual(["test.", "second."], Texts).
+
+multi_node() ->
+    [{doc, "Tests several nodes starting concurrently"}].
+
+multi_node(Config) when is_list(Config) ->
+    Conn = proplists:get_value(connection, Config),
+    Peers = [
+        peer:start_link(#{name => peer:random_name(?FUNCTION_NAME),
+            wait_boot => {self(), tag}, connection => Conn})
+        || _ <- lists:seq(1, 4)],
+    Nodes = [receive {tag, {started, Node, Peer}} -> Node end || {ok, Peer} <- Peers],
+    ?assertEqual(4, length(Nodes)),
+    [?assertEqual(ok, peer:stop(Peer)) || {ok, Peer} <- Peers].
+
+duplicate_name() ->
+    [{doc, "Tests that a node with the same name fails to start"}].
+
+duplicate_name(Config) when is_list(Config) ->
+    Conn = proplists:get_value(connection, Config),
+    {ok, Peer, _Node} = peer:start_link(#{connection => Conn, name => ?FUNCTION_NAME, register => false}),
+    ?assertException(exit, _, peer:start_link(#{connection => standard_io, name => ?FUNCTION_NAME})),
+    peer:stop(Peer).
+
+%% -------------------------------------------------------------------
+%% Compatibility: old releases
+old_release() ->
+    [{doc, "Verity running with previous OTP release"}].
+
+old_release(Config) when is_list(Config) ->
+    PrivDir = proplists:get_value(priv_dir, Config),
+    %% don't expect OTP 10 to be available
+    ?assertEqual(not_available, ?CT_PEER([], "10", PrivDir)),
+    PrevRel = integer_to_list(list_to_integer(erlang:system_info(otp_release)) - 2),
+    case ?CT_PEER([], PrevRel, PrivDir) of
+        not_available ->
+            {skip, "OTP " ++ PrevRel ++ " not found"};
+        {ok, Peer, Node} ->
+            ?assertEqual(PrevRel, rpc:call(Node, erlang, system_info, [otp_release])),
+            peer:stop(Peer)
+    end.
+
+%% -------------------------------------------------------------------
+%% SSH/Docker cases
+
+ssh() ->
+    [{doc, "Tests ssh (localhost) node support"}].
+
+ssh(Config) when is_list(Config) ->
+    %% figure out path to 'erl' locally ('erl' may not be in path via SSH)
+    case {os:find_executable("erl"), os:find_executable("ssh")} of
+        {false, _} ->
+            {skip, "erl not found"};
+        {ErlPath, SshPath} ->
+            Name = peer:random_name(?FUNCTION_NAME),
+            {ok, Peer, _Node} = peer:start_link(#{exec => {SshPath, ["localhost", ErlPath]},
+                connection => standard_io, name => Name,
+                args => ["-env", "BINDIR", os:getenv("BINDIR")], host => "localhost"}),
```

Comment:
@max-au Do you recall why you needed to have the BINDIR here? This testcase fails on windows as BINDIR is not set there so I'm thinking of removing the `-env` bit of this testcase.

Reply:
Ouch, I think it's an accidental leftover from my original attempt to make this test case work without any Erlang installation (when `os:get_executable("erl")` returns `false`). Initial version used a technique similar to `peer:default_erts`, starting `erlexec` with BINDIR transitively set via SSH. Sure, go ahead and remove it.